nucypher/config/base.py

```diff
@@ -0,0 +1,216 @@
+        return filename
+
+    @classmethod
+    def default_filepath(cls) -> str:
+        """
+        Generates the default configuration filepath for the class.
+
+        :return: The generated filepath string
+        """
+        filename = cls.generate_filename()
+        default_path = os.path.join(cls.DEFAULT_CONFIG_ROOT, filename)
+        return default_path
+
+    def generate_filepath(self, filepath: str = None, modifier: str = None, override: bool = False) -> str:
+        """
+        Generates a filepath for saving to writing to a configuration file.
+
+        Default behavior *avoids* overwriting an existing configuration file:
+
+        - The filepath exists and a filename `modifier` is not provided, then `FileExistsError` will be raised.
+        - The modified filepath exists, then `FileExistsError` will be raised.
+
+        To allow re-generation of an existing filepath, set `override` to True.
+
+        :param filepath: A custom filepath to use for configuration.
+        :param modifier: A unique string to modify the filename if the file already exists.
+        :param override: Allow generation of an existing filepath.
+        :return: The generated filepath.
+
+        """
+        if not filepath:
+            filename = self.generate_filename()
+            filepath = os.path.join(self.config_root, filename)
+        if os.path.exists(filepath) and not override:
+            if not modifier:
+                raise FileExistsError(f"{filepath} exists and no filename modifier supplied.")
+            filename = self.generate_filename(modifier=modifier)
+            filepath = os.path.join(self.config_root, filename)
+        self.filepath = filepath
+        return filepath
+
+    def to_configuration_file(self, filepath: str = None, modifier: str = None, override: bool = False) -> str:
+        filepath = self.generate_filepath(filepath=filepath, modifier=modifier, override=override)
+        filepath = self._write_configuration_file(filepath=filepath, override=override)
+        return filepath
+
+    @classmethod
+    def from_configuration_file(cls, filepath: str = None, **overrides) -> 'BaseConfiguration':
+        filepath = filepath or cls.default_filepath()
+        payload = cls._read_configuration_file(filepath=filepath)
+        instance = cls(filepath=filepath, **payload, **overrides)
+        return instance
+
+    @classmethod
+    def _read_configuration_file(cls, filepath: str) -> dict:
+        """Reads `filepath` and returns the deserialized JSON payload dict."""
+        with open(filepath, 'r') as file:
+            raw_contents = file.read()
+            payload = cls.deserialize(raw_contents)
+        return payload
+
+    def _write_configuration_file(self, filepath: str, override: bool = False) -> str:
+        """Writes to `filepath` and returns the written filepath.  Raises `FileExistsError` if the file exists."""
+        if os.path.exists(filepath) and not override:
```

Comment:
What is "override" here?
Equivalent to "fail silently"?

Reply:
`override` allows the file to be written if it exists.